app/views/users/sessions.py

```diff
@@ -141,6 +146,69 @@ def add_speaker_view(event_id, session_id):
         return redirect(url_for('.index_view', event_id=event_id))
 
 
+@event_sessions.route('/download_speakers_as_csv/')
+@belongs_to_event
+@can_access
+def download_speakers_as_csv(event_id):
+    sessions = DataGetter.get_sessions_by_event_id(event_id)
+    event = DataGetter.get_event(event_id)
+    main = [["Speaker Name", "Speaker Session(s)", "Speaker Email", "Speaker Status", \
+             "Speaker Mobile", "Speaker Organisation", "Speaker Position"]]
+    for session in sessions:
+        for speaker in session.speakers:
+            data = [speaker.name if speaker.name else '', session.title if session.title else '',
+                    speaker.email if speaker.email else '']
+            if speaker.sessions:
+                for sess in speaker.sessions:
+                    data.append(sess.state)
```

Comment:
A speaker may have more than one session. Don't you think in that case this will actually mess up the row?? Please test as @niranjan94 suggested.